test/Microsoft.Azure.Devices.Edge.Test.Common/config/BaseModuleConfigBuilder.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public abstract class BaseModuleConfigBuilder : IModuleConfigBuilder
+    {
+        IDictionary<string, object> DesiredProperties { get; }
+
+        IDictionary<string, string> Env { get; }
+
+        protected IDictionary<string, object> Deployment { get; }
+
+        protected IDictionary<string, object> Settings { get; }
+
+        public string Name { get; }
+
+        protected BaseModuleConfigBuilder(string name, string image)
```

Comment:
need to check if name and/or image variables are empty or null.

Reply:
This is internal test code, so I wanted to keep error handling light. Callers need to make sure they don't pass meaningless values to these functions.